lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp

```diff
@@ -226,7 +230,7 @@ LazyNativeFunctions::convolution_backward_overrideable(
   const auto kernel_size = weight.sizes().slice(2);
   CHECK(kernel_size.size() == 2 || kernel_size.size() == 3);
   const at::DeviceType device_type =
-      lazy_tensors::compiler::TSComputationClient::HardwareDeviceType();
+      torch_lazy_tensors::compiler::getBackendRegistrar()->HardwareDeviceType();
```

Comment:
This is not a issue since this code is in `ts_backend` code but xla behaves a bit different than ts in this case.

Even in XLA:GPU and XLA:TPU setup, we always have `CPU:0` as one of the device. In a TPU setup, we have
```
CPU:0, TPU:0-> TPU:7
```
This might be why we use the device carried by each `XLATensor` to determine the hardware type instead of query a process local env var. Although I think in practical use case, we never put any tensor on `CPU:0` if we have `GPU:0` or `TPU:0`.

Reply:
I think we can arrange for a map layer that translates from ordinals related to c10 devices if we need to (so, XLA:0-XLA:N) to real runtime devices (e.g. XLA:0-7 maps to XLA::TPU:0-7, XLA:8 maps to XLA::CPU:0),  so at least there is no need for having multiple different enums in the same software layer.

Is there a need to reason differently about the 'compilation' device of a tensor though?  OR is it possible inside xla backend to take the runtime device of a tensor and then infer the right compilation device?

Do we need to even expose an API from the backend interface into Lazy Core to tell which backend devices are available?  I guess we would want this at minimum for error checking reasons, and possibly also to configure the 'map' I described above.  Alternatively, we could just simplify the API to not expose hardware details, and then pass c10:: device ordinal (XLA:3) down into XLA backend, where it would do some kind of mapping which would be controlled by env variables or system config.  If XLA:3 isn't an available device, it could throw.